File: templates/tracker-template-typescript/example-tracker.ts
type Severity = "low" | "medium" | "high" | "critical";

interface EventRecord {
  eventId: string;
  source: string;
  eventType: string;
  severity: Severity;
  payload: Record<string, unknown>;
}

function shouldNotify(event: EventRecord, minimum: Severity): boolean {
  const order: Record<Severity, number> = { low: 0, medium: 1, high: 2, critical: 3 };
  return order[event.severity] >= order[minimum];
}

const event: EventRecord = {
  eventId: "sample-ts-001",
  source: "template",
  eventType: "sample.update",
  severity: "high",
  payload: { message: "template event" },
};

console.log({ shouldNotify: shouldNotify(event, "medium") });

File: templates/tracker-template-python/example_tracker.py
from src.event_schema import EventRecord, validate_event
from src.rules.base import SeverityAtLeastRule


def run_sample() -> bool:
    event = EventRecord(
        event_id="sample-001",
        source="template",
        event_type="sample.update",
        severity="high",
        payload={"message": "template event"},
    )
    validate_event(event)
    return SeverityAtLeastRule(minimum="medium").evaluate(event)


if __name__ == "__main__":
    print({"should_notify": run_sample()})
